### PR TITLE
Add precondition check util

### PR DESCRIPTION
**Pull request by marekoid**

### CODE REVIEW

File: src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java [hunk 1]
```diff
@@ -33,19 +36,14 @@ a copy of this software and associated documentation files (the "Software"),
     private byte[] key;
 
     public SecretBoxOpener(byte[] key) {
-        // TODO: add a a little preconditions lib/class for that
-        if (key == null) throw new IllegalArgumentException("null key passed");
-        if (key.length != 32) {
-            throw new IllegalArgumentException("key should be 32B, is: " + key.length + "B");
-        }
+        checkNotNull(key, "null key passed");
+        checkArgument(key.length == 32, "key should be 32B, is: " + key.length + "B");
```
Comment (mdpye) [inline src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java]:
```suggestion
        checkArgument(key.length == 32, "key length should be 32 bytes, but is: " + key.length + " bytes");
```

Comment (marekoid) [reply src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java]: not sure what was wrong with the concise form, but didn't mind much the longer one so done that: dbdea4e53d468051462e3e7480e7a0a5368d174d just couldn't apply from GH as the 100 char limit was broken

File: src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java [hunk 2]
```diff
@@ -33,19 +36,15 @@ a copy of this software and associated documentation files (the "Software"),
     private byte[] key;
 
     public SecretBoxOpener(byte[] key) {
-        // TODO: add a a little preconditions lib/class for that
-        if (key == null) throw new IllegalArgumentException("null key passed");
-        if (key.length != 32) {
-            throw new IllegalArgumentException("key should be 32B, is: " + key.length + "B");
-        }
+        checkNotNull(key, "null key passed");
+        checkArgument(key.length == 32, "key length should be 32 bytes, but is " +
```
Comment (daniellevass) [inline src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java]: nit: should -> must if we're failing people for it not being what we expect.

Comment (marekoid) [reply src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java]: I'm not sure if that doesn't sound too strong and if "should" is not more common for such messages (after all it also can be used to express obligation). @mdpye thoughts? Changed it for feedback: bb925cd

**Pull request merged**